parachain/primitives/core/src/outbound.rs

```diff
@@ -129,18 +141,68 @@ impl Command {
 		}
 	}
 
+	/// Compute gas cost
+	/// reference gas from benchmark report with some extra margin for incentive
+	/// | Function Name    | min    | avg    | median | max    | # calls |
+	/// | agentExecute    | 487    | 5320   | 3361   | 14074  | 4       |
+	/// | createAgent    | 839    | 184709 | 237187 | 237187 | 9       |     
+	/// | createChannel    | 399    | 31023  | 2829   | 75402  | 5       |
+	/// | updateChannel    | 817    | 15121  | 3552   | 36762  | 5     |     
+	/// | transferNativeFromAgent    | 770    | 21730  | 21730  | 42691  | 2       |
+	/// | setOperatingMode    | 682    | 12838  | 13240  |  24190  | 4       |
+	/// | upgrade    | 443    | 9270   | 3816   | 29004  | 4       |
+	pub fn dispatch_gas(&self) -> u128 {
+		match self {
+			Command::CreateAgent { .. } => 300000,
+			Command::CreateChannel { .. } => 100000,
+			Command::UpdateChannel { .. } => 50000,
+			Command::TransferNativeFromAgent { .. } => 60000,
+			Command::SetOperatingMode { .. } => 40000,
+			Command::AgentExecute { command, .. } => match command {
+				AgentExecuteCommand::TransferToken { .. } => 30000,
+			},
+			// leave enough space for upgrade with arbitrary initialize logic
+			Command::Upgrade { .. } => 300000,
+		}
+	}
+
+	pub fn base_fee_required(&self) -> bool {
+		match self {
+			Command::CreateAgent { .. } => true,
+			Command::CreateChannel { .. } => true,
+			Command::AgentExecute { .. } => true,
+			Command::UpdateChannel { .. } => true,
+			Command::TransferNativeFromAgent { .. } => true,
+			Command::Upgrade { .. } => false,
+			Command::SetOperatingMode { .. } => false,
+		}
+	}
+
+	pub fn extra_fee_required(&self) -> bool {
+		match self {
+			Command::CreateAgent { .. } => true,
+			Command::CreateChannel { .. } => true,
```

Comment:
Why is an extra fee required for CreateAgent and CreateChannel. I was under the impression that these commands have a fixed cost?

Reply:
The meaning of "extra fee" in this context is that the sender needs to pay the full cost of execution on the Ethereum side.

I agree the terminology is confusing and we should settle on better naming :)

Reply:
https://github.com/Snowfork/snowbridge/pull/940#discussion_r1333099420